lib/Doctrine/DBAL/Schema/ColumnDiff.php

```diff
@@ -77,6 +77,9 @@ public function hasChanged($propertyName)
      */
     public function getOldColumnName()
     {
+        if ($this->column->isQuoted()) {
```

Comment:
This logic doesn't quite make sense. You're saying that if the 'to' column name is quoted, the 'from' column name should be as well. 

The issue is really that in Comparator, the ColumnDiff is created like this:
```php
new ColumnDiff($column->getName(), $table2->getColumn($columnName), $changedProperties)
```

Which results in the 'from' column name always being unquoted, because even though the from column may originally have been properly marked as quoted, that information is not being passed into the ColumnDiff constructor. I see the same sort of issue in a few other places. It needs a more comprehensive fix than this, as this simply hides the issue.